core/src/utils/periodic_task_test.cpp

```diff
@@ -418,4 +418,39 @@ UTEST_F(PeriodicTaskLog, ErrorLog) {
     task.Stop();
 }
 
+UTEST(PeriodicTask, EnableDisable) {
```

Comment:
There is a lot to decipher in this test's logic - split it into some steps with empty lines and comment on each step